unmock/core/options.py

```diff
@@ -41,6 +46,45 @@ def __init__(self, save: Union[bool, List[str]] = False, unmock_host: str = UNMO
             persistence = FSPersistence(self.token)
         self.persistence = persistence
 
+    def get_token(self) -> Optional[str]:
+        """
+        Fetches a new access token from the unmock server or predisposed access token if it is still valid.
+        Throws RuntimeError on logical failures with unexpected responses from the Unmock host.
+        """
+        url = "{scheme}://{host}:{port}".format(scheme=self.scheme, host=self.unmock_host, port=self.unmock_port)
+        access_token = self.persistence.load_auth()
+        if access_token is not None:  # If we already have an access token, let's see we can still ping with it
+            if self._validate_access_token(access_token):  # We can ping, all's good in the world!
+                return access_token
+
+        # Otherwise, we need to get a new token
+        refresh_token = self.persistence.load_token()
+        if refresh_token is None:
+            return  # Continue with the public API ('/y/' version)
+        response = requests.post("{url}/token/access".format(url=url), json={"refreshToken": refresh_token})
+        if response.status_code == HTTPStatus.OK:
```

Comment:
Same as `res.ok`?

Reply:
Unfortunately not, `res.ok` has ```Returns True if :attr:`status_code` is less than 400, False if not.```

Reply:
Ah ok. Generally I'd assume any 2xx should be considered "ok", for example 204 is just as valid response for `/ping` as 200 and any client should not rely on explicit 200 imho. But it does not matter ofc that much here if `/token/access` promises to return 200.

Reply:
2xx might be a good idea; but for example 3xx would still mean an error IMO.
We should just align the responses from the `unmock-cloud`, so we have a contract between the two :)